articles/how-to-use-streams-in-node,js/index.md

```diff
@@ -0,0 +1,116 @@
+### Introduction
+Streams are objects that allow one to read data from a source or write to a destination in a continuous manner. Node.js has four types of streams; readable, writable, duplex and transform. Each stream is an `eventEmitter` instance that emits different events at several intervals. This article goes through reading and writing data to files using streams.
+
+### Prerequisites
+To follow through this article, you'll need to have node.js installed and have a basic knowledge of Javascript.
+
+Let's get started!
+
+What do the different streams do? 
+ 1. The readable stream is a stream that is used for read operations.
+ 2. The Writable stream as the name suggests is a stream used for write operations.
+ 3. A duplex stream is used for both read and write operations. Lastly,
+ 4. The transform stream is a type of stream that computes the output based on the input.
+
+The streams throw several events since they are `eventEmitter` instances. These events are used to track and monitor the stream. Some of the most commonly used events are;
+ 1. Data - This event is fired when readable data is available.
+ 2. Finish - This event is fired when the stream is done writing data.
+ 3. Error - This event is fired when any error occurs while reading or writing data.
+ 4. End - This event is fired when the stream is done reading data.
+
+With this information, let's have a look at how we can use a stream to read/write data from/to a file.
+
+### Reading from a stream
+Handling streams is quite simple, all we have to do is create the stream and handle the events emitted by the stream. Let's see how;
+
+First create a folder with a name of your choice. Then create a text file inside the folder named "input.txt".
+
+Add the following text in the file,
+```
+Section partners with university computer science programs across the globe to create a community-generated pool of content which is useful for engineers of every (any) level.
+```
+
```

Comment:
Please inform the readers beforehand that they should initialise the nodejs project otherwise it would not work.

Reply:
Thank you for the suggestion @LinusMuema. It actually works without initializing the project. The node.js runtime is all that is needed.